### PR TITLE
Improve error message when GroupReadsByUmi produces out of order reads.

### PR DESCRIPTION
**Pull request by nh13**
See:
- https://github.com/fulcrumgenomics/fgbio/issues/413
- https://github.com/fulcrumgenomics/fgbio/issues/416
- https://github.com/fulcrumgenomics/fgbio/issues/394

### CODE REVIEW

File: src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala [hunk 1]
```diff
@@ -514,6 +514,20 @@ class GroupReadsByUmi
     }
   }
 
+  /** The error message when reads are out of order. */
+  private def getReadsOutOfOrderErrorMessage(r1: SamRecord, r2: SamRecord): String = {
+    val errorMessage = if (this.rawTag == "MI") {
+      s"Using 'MI' tag as the 'raw tag', are you sure this is correct?  Did you mean the 'RX' tag?"
```
Comment (tfenne) [inline src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala]: Should we just check this at startup and emit a warning?

File: src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala [hunk 2]
```diff
@@ -514,6 +514,20 @@ class GroupReadsByUmi
...
+    }
+    else if (this.rawTag == this.assignTag) {
```
Comment (tfenne) [inline src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala]: This should, I think, be a validation error at startup.  I don't think we should allow this.

File: src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala (hunk 1 above)
Comment (fleharty) [reply src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala]: @tfenne What UmiAwareMarkDuplicatesWithMateCigar is doing does appear to be very wrong based on what you've shown.  Could you share that bam, I would like to try to fix this issue in Picard.

Comment (tfenne) [reply src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala]: @fleharty I don't think I can share that BAM, but I can probably use what I figured out to generate a small test case that I can pass along.  I'll try and do that in the next couple of days.

Comment (fleharty) [reply src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala]: @tfenne I just wanted to follow up on this.  Could you generate the small test case?
Thanks!

File: src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala (hunk 2 above)
Comment (nh13) [reply src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala]: Done

**Comment by tfenne** [comment]
I think the upshot of this PR is that the problem lies elsewhere.  I'm closing this PR because I don't think it solves the stated problem, that problem lies in another program not giving the same `MI` value to both ends of a pair.

**Comment by fleharty** [comment]
@tfenne If you get a chance could you show me an example of where this happens, I've not seen this happen so if there is an example somewhere I may be able to fix it.

**Pull request closed without merging**